SDK_dotnet/GluwaClient.cs

```diff
@@ -0,0 +1,402 @@
+
+            var queryParams = new List<string>();
+            if (offset > 0)
+            {
+                queryParams.Add($"offset={offset}");
+            }
+            if (limit > 0)
+            {
+                queryParams.Add($"limit={limit}");
+            }
+            if (queryParams.Any())
+            {
+                queryParams.Add($"status={status}");
+                requestUri = $"{requestUri}?{string.Join("&", queryParams)}";
+            }
+
+            using (HttpClient httpClient = new HttpClient())
+            {
+                httpClient.DefaultRequestHeaders.Add("X-REQUEST-SIGNATURE", getTimestampSignature());
+                HttpResponseMessage message = await httpClient.GetAsync(requestUri);
+                if (message.IsSuccessStatusCode)
+                {
+                    List<TransactionResponse> transactionResponse = await message.Content.ReadAsAsync<List<TransactionResponse>>();
+                    result.IsSuccess = true;
+                    result.Data = transactionResponse;
+
+                    return result;
+                }
+                else
+                {
+                    throw new HttpListenerException((int)message.StatusCode, message.ReasonPhrase);
+                }
+            }
+        }
+
+        public async Task<Result<TransactionResponse>> GetListTransactionDetailAsync(ECurrency? currency, string txnHash)
+        {
+            if (currency == null)
+            {
+                throw new ArgumentNullException(nameof(currency));
+            }
+            else if (string.IsNullOrEmpty(txnHash))
+            {
+                throw new ArgumentNullException(nameof(txnHash));
+            }
+            var result = new Result<TransactionResponse>();
+
+            string requestUri = $"{mBaseUrl}/v1/{currency}/Transactions/{txnHash}";
+
+            using (HttpClient httpClient = new HttpClient())
+            {
+                httpClient.DefaultRequestHeaders.Add("X-REQUEST-SIGNATURE", getTimestampSignature());
+                HttpResponseMessage message = await httpClient.GetAsync(requestUri);
+                if (message.IsSuccessStatusCode)
+                {
+                    TransactionResponse transactionResponse = await message.Content.ReadAsAsync<TransactionResponse>();
+                    result.IsSuccess = true;
+                    result.Data = transactionResponse;
+
+                    return result;
+                }
+                else
+                {
+                    throw new HttpListenerException((int)message.StatusCode, message.ReasonPhrase);
+                }
+            }
+        }
+
+        public async Task<bool> PostTransactionAsync(
+            ECurrency? currency,
+            string amount,
+            string target,
+            string merchantOrderID = "",
+            string note = "",
+            int expiry = 1800
```

Comment:
Because of my mistake, I changed the default value to 0. By the way, if you look at the SDK for php, there is an expiry in the parameter when creating a transaction. I do not look necessary. Am I wrong?
<img width="479" alt="createTransaction" src="https://user-images.githubusercontent.com/50939818/70782604-bfff4b80-1dca-11ea-9c3e-53a6e4078f7d.png">

Reply:
You need to pass in the expiry when creating the transaction if the user provided. The current code you have looks correct.